src/main/host/syscall/socket.c

```diff
@@ -310,40 +300,38 @@ static int _syscallhandler_getTCPOptHelper(SysCallHandler* sys, TCP* tcp,
     }
 }
 
-static int _syscallhandler_getSocketOptHelper(SysCallHandler* sys, Socket* sock,
-                                              int optname, PluginPtr optvalPtr,
-                                              PluginPtr optlenPtr) {
-    /* Get the len via clone, so we can write to optlenPtr too. */
-    const socklen_t* optlen =
-        process_getReadablePtr(sys->process, sys->thread, optlenPtr, sizeof(*optlen));
-
+static int _syscallhandler_getSocketOptHelper(SysCallHandler* sys, Socket* sock, int optname,
+                                              void* optval, socklen_t* optlen) {
     /* All options we currently support are integers. When adding support for
      * more options, make sure to check length requirements. */
```

Comment:
I think this comment needs updating. It looks like this change is meant to do one or both of making it easier to support non-int-sized options in the future, and in the current options handle if the caller passes an integer type smaller than `int`? For the latter, is that actually something we need to support?

Reply:
The goal was just to make it behave more like Linux, where the program can call getsockopt with any length, and it will only copy that many bytes. We don't need to support that, and I can't think of any legitimate reasons a program would do that, but changed it just so that it better follows the Linux system call interface and since we can do it safely. When writing tests, it's useful to test things like passing small buffers to make sure Shadow doesn't segfault. But if Shadow behaves different (even if there isn't a bug/segfault/etc), than we need to test for two different outcomes depending on whether the test is running under Shadow or not, and figured it would just be easier to make Shadow behave the same as Linux. But if there are any downsides to this change let me know and I'm okay with changing it back.

And the sizes for the implemented options in `_syscallhandler_getSocketOptHelper()` are only ints, but `_syscallhandler_getTCPOptHelper()` has an option which returns a `struct tcp_info`, which is much larger, so it's nice to have the `process_getWriteablePtr()` in the parent function instead of duplicating it in both helpers.

Reply:
Yup, seems reasonable. Would still be good to update the comment, though, since it still reads as if the size were hard-coded to be an integer. Maybe we can just be dropped entirely?